mypy/stubtest.py

```diff
@@ -985,16 +987,16 @@ def verify_typealias(
     }
 )
 
-ALLOW_MISSING_CLASS_DUNDERS = frozenset(
+IGNORABLE_CLASS_DUNDERS = frozenset(
     {
         # Special attributes
         "__dict__",
         "__text_signature__",
         "__weakref__",
         "__del__",  # Only ever called when an object is being deleted, who cares?
-        # These two are basically useless for type checkers
         "__hash__",
         "__getattr__",  # resulting behaviour might be typed explicitly
+        "__setattr__",  # defining this on a class can cause worse type checking
```

Comment:
Hmm, this one seems to be rare enough that I think I'd prefer to consider each case separately and allowlist them in typeshed. I can't think of an example where including it in the stub would improve type checking, but you never know?